### PR TITLE
[Merged by Bors] - feat: Condensed objects

### PR DESCRIPTION
**Pull request by adamtopaz**
---
- [x] depends on: #4505 

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by semorrison** [comment]
This PR/issue depends on:
* leanprover-community/mathlib4#4505
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

File: Mathlib/Topology/Category/CompHaus/EffectiveEpi.lean [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+
+open EffectiveEpiFamily
+
+theorem effectiveEpiFamily_tfae
+    {α : Type} [Fintype α] {B : CompHaus.{u}}
+    (X : α → CompHaus.{u}) (π : (a : α) → (X a ⟶ B)) :
+    [ EffectiveEpiFamily X π
+    , Epi (Sigma.desc π)
+    , ∀ b : B, ∃ (a : α) (x : X a), π a x = b
+    ].TFAE := by
```
Comment (semorrison) [inline Mathlib/Topology/Category/CompHaus/EffectiveEpi.lean:174]: Suggest writing `List.TFAE`, rather than using dot notation, for readability!

Comment (adamtopaz) [reply Mathlib/Topology/Category/CompHaus/EffectiveEpi.lean:174]: Good idea! Done.

File: Mathlib/Condensed/Basic.lean [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+## Implementation Details
+
+We use the coherent Grothendieck topology on `CompHaus`, and define condensed objects in `C` to
+be `C`-valued sheaves, with respect to this Grothendieck topology.
+
+In future work, we will define similar Grothendieck topologies on the category of profinite sets
+and extremally disconnected sets, and show that the three categories are equivalent (under
+suitable assumptions on `C`).
+
+Note: Our definition more closely resembels "Pyknotic objects" in the sense of Barwick-Haine,
```
Comment (jcommelin) [inline Mathlib/Condensed/Basic.lean:25]:
```suggestion
Note: Our definition more closely resembles "Pyknotic objects" in the sense of Barwick-Haine,
```

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover-community/mathlib4#4505~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

File: Mathlib/Topology/Category/CompHaus/ExplicitLimits.lean [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+union with its usual topology.
+-/
+def finiteCoproduct : CompHaus := CompHaus.of <| Σ (a : α), X a
+
+/--
+The inclusion of one of the factors into the explicit finite coproduct.
+-/
+def finiteCoproduct.incl (a : α) : X a ⟶ finiteCoproduct X where
+  toFun := fun x => ⟨a,x⟩
+  continuous_toFun := continuous_sigmaMk (σ := fun a => X a)
```
Comment (semorrison) [inline Mathlib/Topology/Category/CompHaus/ExplicitLimits.lean:129]: Maybe still name this `\iota` to match the limits library? Not essential.

Comment (adamtopaz) [reply Mathlib/Topology/Category/CompHaus/ExplicitLimits.lean]: Done!

**Review by kim-em — APPROVED** [review]

**Comment by jcommelin** [comment]
Thanks :tada:

bors merge

**Pull request closed without merging**